segmentstore/server/src/main/java/io/pravega/segmentstore/server/writer/AppendIntegrityChecker.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.segmentstore.server.writer;
+
+import io.pravega.common.util.BufferView;
+import io.pravega.segmentstore.server.DataCorruptionException;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.concurrent.GuardedBy;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Queue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+/**
+ * Helper class to check internal data integrity of Appends within Pravega ingestion pipeline. The main purpose of this
+ * class is to compute and keep track of hashes of Append contents upon ingestion and then verify that the integrity of
+ * such Appends at the end of the ingestion pipeline (when data is moved to LTS).
+ */
+@Slf4j
+public class AppendIntegrityChecker implements AutoCloseable {
+    //region Members
+
+    // Represents that no hash has been computed for a given Append.
+    public static final long NO_HASH = Long.MIN_VALUE;
+
+    @GuardedBy("this")
+    private final Queue<AggregatedAppendIntegrityInfo> appendIntegrityInfo;
+
+    //endregion
+
+    //region Constructor
+
+    public AppendIntegrityChecker() {
+        this.appendIntegrityInfo = new LinkedBlockingQueue<>();
+    }
+
+    //endregion
+
+    //region AppendIntegrityChecker Implementation
+
+    public static long computeDataHash(BufferView data) {
+        return computeDataHash(data.getCopy());
+    }
+
+    public static long computeDataHash(byte[] data) {
+        return Arrays.hashCode(data);
```

Comment:
Use `BufferView.hashCode`. It's vectorized and will save you all the trouble of having to make a copy just for computing the hash. Much faster.

Reply:
Check out the implementation. It has all the answers you need. In terms of performance, I do not see a faster way of doing it compared to how it is done now. Short of using actual SIMD instructions, the kind of vectorization used in the BufferView implementations is as fast as you're going to get here.

Reply:
Using PR #6842 for computing hash.